controllers/machinehealthcheck_controller.go

```diff
@@ -254,10 +255,50 @@ func (r *MachineHealthCheckReconciler) reconcile(ctx context.Context, logger log
 
 		if annotations.IsPaused(cluster, t.Machine) {
 			logger.Info("Machine has failed health check, but machine is paused so skipping remediation", "target", t.string(), "reason", condition.Reason, "message", condition.Message)
-		} else {
+		} else if m.Spec.RemediationTemplate != nil {
+
+			cloneOwnerRef := &metav1.OwnerReference{
+				APIVersion: clusterv1.GroupVersion.String(),
+				Kind:       t.Machine.Kind,
+				Name:       t.Machine.Name,
+				UID:        t.Machine.UID,
+			}
+
+			from, err := external.Get(ctx, r.Client, m.Spec.RemediationTemplate, t.Machine.Namespace)
+			if err != nil {
+				return ctrl.Result{}, errors.Wrapf(err, "Remediation template not found %q", m.Spec.RemediationTemplate)
+			}
+
+			generateTemplateInput := &external.GenerateTemplateInput{
+				Template:    from,
+				TemplateRef: m.Spec.RemediationTemplate,
+				Namespace:   t.Machine.Namespace,
+				ClusterName: t.Machine.ClusterName,
+				OwnerRef:    cloneOwnerRef,
+				Labels:      nil,
+			}
+			to, err := external.GenerateTemplate(generateTemplateInput)
+			if err != nil {
+				return ctrl.Result{}, errors.Wrapf(err, "Failed to create template for remediation request %q", m.Spec.RemediationTemplate)
+			}
+
+			to.SetName(t.Machine.Name)
+
+			// Create the external clone.
+			if err := r.Client.Create(context.Background(), to); err != nil {
+				return ctrl.Result{}, errors.Wrapf(err, "Failed to create remediation request for machine %q", t.Machine.Name)
+			}
+
+			remediationRef := external.GetObjectReference(to)
+
+			t.Machine.Status.RemediationRef = remediationRef
```

Comment:
If we fail to save the machine status later (line 302), we will end up with a machine without `RemediationRef` but with Remediation Request CR.
On next reconcile loops, we will fail on `Create()` as the CR already exists and we'll never reach this line again.
This basically means that this CR will never get deleted until machine goes healthy and unhealthy again, as we we only delete the CR if `RemediationRef` exists.